marlowe-playground-client/src/Marlowe/Holes.purs

```diff
@@ -125,30 +128,23 @@ getMarloweConstructors BoundType = Map.singleton "Bound" [ DataArg "from", DataA
 
 getMarloweConstructors TokenType = Map.singleton "Token" [ DataArg "currency", DataArg "token" ]
 
+-- | Creates a String consisting of the data constructor name followed by argument holes
+--   e.g. "When [ ?case_10 ] ?timeout_11 ?contract_12"
 constructMarloweType :: String -> MarloweHole -> Map String (Array Argument) -> String
 constructMarloweType constructorName (MarloweHole { name, marloweType, start }) m = case Map.lookup constructorName m of
   Nothing -> ""
   Just [] -> constructorName
-  Just vs ->
-    let
-      Position { line, column } = start
-    in
-      "(" <> constructorName <> " " <> intercalate " " (mapWithIndex (showArgument line column) vs) <> ")"
+  Just vs -> "(" <> constructorName <> " " <> intercalate " " (mapWithIndex (showArgument start) vs) <> ")"
   where
-  showArgument line column i (ArrayArg arg) = "[ ?" <> arg <> "_" <> show line <> "_" <> show (column + i) <> " ]"
+  showArgument startPos i (ArrayArg arg) = "[ ?" <> arg <> "_" <> show (startPos + i) <> " ]"
 
-  showArgument line column i (DataArg arg) = "?" <> arg <> "_" <> show line <> "_" <> show (column + i)
+  showArgument startPos i (DataArg arg) = "?" <> arg <> "_" <> show (startPos + i)
 
-  showArgument _ _ _ NewtypeArg = ""
-
-derive instance genericMarloweType :: Generic MarloweType _
-
-instance showMarloweType :: Show MarloweType where
-  show = genericShow
+  showArgument _ _ NewtypeArg = ""
 
 data Term a
   = Term a
-  | Hole String (Proxy a) Position Position
+  | Hole String (Proxy a) Pos Pos
```

Comment:
Should this be a record? _ducks_

Reply:
maybe in a different PR because I'll be looking into it more.